tests/jenkins/.gitignore

```diff
@@ -0,0 +1,7 @@
+__pycache__
```

Comment:
The .gitignore in the root of the repo makes most of this redundant, and I believe even the "results" entry isn't needed, since people won't be running the Jenkins tests locally (and a dirty git directory doesn't matter in a one-off CI environment that gets deleted at end of run).